Dockerfile.template

```diff
@@ -32,3 +32,12 @@ RUN curl -sSL "https://github.com/gotestyourself/gotestsum/releases/download/v${
 
 ENV GOPATH /home/circleci/go
 ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
+
+# Install govulncheck on a temporary stage to ensure minimal size footprint
+# by copying to output layer just the govulncheck binary
+FROM base AS govulncheck
```

Comment:
Reason why I've chosen to use multistage approach is to simplify the clean up. `go install` pulls dependencies, compiles them. This populates go mod and compiler build caches. This can be replaced with `go clean -cache -modcache && rm -rf "${GOPATH}/pkg"` but I am not 100% sure that this is the correct approach.

Reply:
Please let me know what you think of these 2 points

Reply:
I think it's better here to use `go clean -cache -modcache && rm -rf "${GOPATH}/pkg"` at the end and just install whatever is needed without a multistage build. This will make it easier if other `go install` based tools are also installed.

Reply:
@felicianotech addressed your remarks on https://github.com/CircleCI-Public/cimg-go/pull/242/commits/871a7192857023af1430ea47e9d007817d480a68